src/main/java/com/zufar/onlinestore/security/configuration/UserIdMdcFilter.java

```diff
@@ -0,0 +1,46 @@
+package com.zufar.onlinestore.security.configuration;
+
+import com.zufar.onlinestore.security.api.SecurityPrincipalProvider;
+import com.zufar.onlinestore.security.jwt.filter.JwtAuthenticationProvider;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.AllArgsConstructor;
+import org.slf4j.MDC;
+import org.springframework.core.annotation.Order;
+import org.springframework.lang.NonNull;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+
+/**
+ * Enriches MDC with user.userId.
+ */
+@Order
```

Comment:
why do we decide to use the most lower precedence here?

Reply:
To make sure it invokes after security filters